library/ssl_misc.h

```diff
@@ -1932,37 +1932,37 @@ int mbedtls_ssl_tls13_handshake_server_step(mbedtls_ssl_context *ssl);
 static inline unsigned mbedtls_ssl_conf_tls13_check_kex_modes(mbedtls_ssl_context *ssl,
                                                               int kex_mode_mask)
 {
-    return (ssl->conf->tls13_kex_modes & kex_mode_mask) != 0;
+    return (ssl->conf->tls13_kex_modes & kex_mode_mask) == 0;
```

Comment:
This still doesn't conform to our [coding standards](https://mbed-tls.readthedocs.io/en/latest/kb/development/mbedtls-coding-standards/#return-type): a `check` function should return an **error code** if the check fails, i.e. a negative value. Typically `check` functions return an appropriate error code, e.g. `INVALID_ARGUMENT` for a validity check or `NOT_PERMITTED` for a permission check.

I think for this function it would make more sense to keep the boolean result (1 if the modes are present in the mask, otherwise 0), and change the name of the function.

A good name would also make it clear whether the function requires _at least one_ or _all_ modes to be present. So something like `mbedtls_ssl_conf_tls13_is_kex_mode_enabled`.

Reply:
Also the function should return `int` either way (either a boolean `int` for an `is` function, or a zero-or-MBEDTLS_ERR_XXX `int` for a `check` function.

Reply:
Do you think we should do the same renaming on `mbedtls_ssl_tls13_check_kex_modes` and `mbedtls_ssl_session_check_ticket_flags`?

Reply:
I guess so, yes. Also it would be useful to have documentation for these functions. When I look at `ssl_misc.h`, I have no idea what `mbedtls_ssl_session_check_ticket_flags` does. It checks something about session tickets, but what? But since it's a boolean result, it should be called something like `mbedtls_ssl_session_has_no_flags` or whatever is being tested.